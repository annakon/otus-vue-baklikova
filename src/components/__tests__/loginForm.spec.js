import {describe, it, expect, beforeEach, vi} from "vitest";
import { mount } from "@vue/test-utils";
import component from "@/components/login/loginForm.vue";

vi.mock('vue-router')
describe("loginForm", () => {
    let wrapper

    beforeEach(() => {
        wrapper = mount(component, { global: {
                stubs: ['RouterLink'],
            } });
    })

    it("mounts without errors", () => {
        expect(wrapper).toBeDefined();
        expect(wrapper.exists()).toBe(true);
    });
    it("renders all", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
