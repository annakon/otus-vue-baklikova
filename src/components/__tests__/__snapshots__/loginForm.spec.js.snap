// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`loginForm > renders all 1`] = `
"<form data-v-12794ea3=\\"\\" novalidate=\\"\\">
  <h1 data-v-12794ea3=\\"\\">Войти</h1>
  <div data-v-12794ea3=\\"\\" class=\\"mb-3 row\\"><label data-v-12794ea3=\\"\\" for=\\"exampleInputEmail1\\" class=\\"col-sm-2 col-form-label\\">логин</label>
    <div data-v-12794ea3=\\"\\" class=\\"col-sm-8\\"><input data-v-12794ea3=\\"\\" type=\\"text\\" class=\\"form-control\\" id=\\"exampleInputEmail1\\" name=\\"login\\">
      <!---->
    </div>
  </div>
  <div data-v-12794ea3=\\"\\" class=\\"mb-3 row\\"><label data-v-12794ea3=\\"\\" for=\\"exampleInputPassword1\\" class=\\"col-sm-2 col-form-label\\">пароль</label>
    <div data-v-12794ea3=\\"\\" class=\\"col-sm-8\\"><input data-v-12794ea3=\\"\\" class=\\"form-control\\" type=\\"password\\" id=\\"exampleInputPassword1\\" name=\\"password\\">
      <!---->
    </div>
    <div data-v-12794ea3=\\"\\" class=\\"mb-3\\">
      <router-link-stub data-v-12794ea3=\\"\\" to=\\"/\\" class=\\"btn btn-primary\\"></router-link-stub>
    </div>
  </div>
</form>"
`;
